lib/auth/tls_test.go

```diff
@@ -2460,8 +2467,8 @@ func (s *TLSSuite) TestTLSFailover(c *check.C) {
 
 	// couple of runs to get enough connections
 	for i := 0; i < 4; i++ {
-		_, err = client.GetDomainName()
-		c.Assert(err, check.IsNil)
+		_, err = client.Get(ctx, client.Endpoint("not", "exist"), url.Values{})
```

Comment:
The per-request failover feature is only present in the HTTP client. The gRPC client fails-over between multiple addresses on instantiating the client, and then continues to use that same address for all requests. 

I could have switched this test to use a different client method that is still HTTP only, but it feels like this test would need to be repeatedly updated as we migrate more API endpoints to gRPC, so I've instead gone for trying to GET a non existent endpoint which will always return a NotFound.